### DISCUSSION TITLE
Release v2.0.0-Beta4

### DISCUSSION THREAD

**Discussion by carlosthe19916** [project-openubl/ublhub#249] [Announcements]
## Changelog

d96e63f 🏁 Releasing version 2.0.0-Beta4
b80cca8 Upgrade xbuilder and xsender dependencies (#248)
e23167b Add cypress (#244)
398b69b ⬆️ Next version 2.0.0-SNAPSHOT

<hr /><em>This discussion was created from the release <a href='https://github.com/project-openubl/ublhub/releases/tag/v2.0.0-Beta4'>Release v2.0.0-Beta4</a>.</em>

**Comment by jwcaceres78** [comment]
Hola con todos, antes que nada felicitar por el excelente trabajo, he estado probado el funcionamiento y tengo un par de dudas:

1. En la documentación indica que puedo utilizar un solo proyecto para varios RUC, mi pregunta es: cómo se configura el usuario y contraseña para el envío de comprobantes, sabiendo que para cada RUC estos datos varían?
2. La segunda consulta es si hay manera de restringir el acceso al front end para que puedan acceder solo usuarios registrados, ya que al momento de levantar el servicio y habilitar el puerto en un VPS este queda expuesto a cualquier persona
3. y la tercera pregunta es si se puede poner como un servicio del sistema operativo, ya que al reiniciar el servidor hay que iniciar nuevamente el servicio de manera manual.

Gracias.

**Reply by user0608** [reply]
Puedes usar este servicio `https://github.com/oauth2-proxy/oauth2-proxy`, es integrable con Google, Kong Gateway también es una opción `https://github.com/Kong/kong`

**Comment by carlosthe19916** [comment]
Hola @jwcaceres78 
> En la documentación indica que puedo utilizar un solo proyecto para varios RUC, mi pregunta es: cómo se configura el usuario y contraseña para el envío de comprobantes, sabiendo que para cada RUC estos datos varían?

1. En efecto, Ublhub puede almacenar XMLs de diferentes RUCs en un solo proyecto. La configuracion SUNAT (usuario, constraseña, URLs) para todos los RUC se configura en el proyecto. Este es un caso para empresas que son PSE (proveedor de servicios electronios) donde un solo certificado, usuario, contraseña, URLs pueden ser usados para multiples empresas (RUCs). En caso de que quieras diferentes usuarios, contraseñas y URLs para cada RUC entonces puedes crear empresas dentro de cada Proyecto en UBLhub

> La segunda consulta es si hay manera de restringir el acceso al front end para que puedan acceder solo usuarios registrados, ya que al momento de levantar el servicio y habilitar el puerto en un VPS este queda expuesto a cualquier persona
2. UBLHub por el momento no tiene implementada la capa de seguridad, seguimos trabajando.

> y la tercera pregunta es si se puede poner como un servicio del sistema operativo, ya que al reiniciar el servidor hay que iniciar nuevamente el servicio de manera manual.

3. Hacer que un programa funcione como servicio en el sistema operativo es muy facil pero depende del sistema operativo que uses, y lo puedes hacer tu mismo, por ejemplo para Windows puedes hacer algo como esto https://stackoverflow.com/questions/415409/run-batch-file-as-a-windows-service . Para Linux tambien hay un monton de tutoriales online.
